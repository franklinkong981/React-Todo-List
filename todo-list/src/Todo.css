.Todo {
  margin: 10px auto;
  font-size: 24px;
  background: rgba(255, 255, 255, 0.5);

  display: flex;
  justify-content: center;
}


.Todo-toggle-button, .Todo-remove-button {
  font-size: 12px;
  margin: 20px;
}

.Todo-toggle-button {
  background-color: yellow;
}

.Todo-remove-button {
  background-color: red;
}