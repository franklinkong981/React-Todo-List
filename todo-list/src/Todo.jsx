import react from "react";

const Todo = ({id, content, completed, removeTodoFunction, toggleTodoFunction}) => {
  const handleRemove = () => {
    removeTodoFunction(id);
  };

  const handleToggle = () => {
    toggleTodoFunction(id);
  };

  const textDecorationStyle = completed ? "line-through" : "none"

  return (
    <div className="Todo" id={id}>
      <h4 className="Todo-text" style={{textDecoration: textDecorationStyle}}>{content}</h4>
      <button className="Todo-toggle-button" onClick={handleToggle}>{completed ? "Unmark as incomplete" : "Mark as complete"}</button>
      <button className="Todo-remove-button" onClick={handleRemove}>Delete Todo</button>
    </div>
  );
};

export default Todo;