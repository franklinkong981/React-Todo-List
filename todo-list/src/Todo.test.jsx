import React from "react";
import {render} from "@testing-library/react";
import Todo from "./Todo.jsx";

const removeTodo = () => {
  console.log("Removed todo");
};

const toggleTodo = () => {
  console.log("Toggled todo");
};

it("renders without crashing", function() {
  render(<Todo id="1" content="Pass the test" completed={false} removeTodoFunction={removeTodo} toggleTodoFunction={toggleTodo}/>);
});

it("matches snapshot", function() {
  const {asFragment} = render(<Todo id="1" content="Pass the test" completed={false} removeTodoFunction={removeTodo} toggleTodoFunction={toggleTodo}/>);
  expect(asFragment()).toMatchSnapshot();
});