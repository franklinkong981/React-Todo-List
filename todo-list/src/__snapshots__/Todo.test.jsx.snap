// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    id="1"
  >
    <p
      class="Todo-text"
      style="text-decoration: none;"
    >
      Pass the test
    </p>
    <button
      class="Todo-toggle-button"
    >
      Mark as complete
    </button>
    <button
      class="Todo-remove-button"
    >
      Delete Todo
    </button>
  </div>
</DocumentFragment>
`;
